### ISSUE TITLE
tf.__version__ raises an AttributeError exception

### ISSUE THREAD

**Issue by ageron** [tensorflow/tensorflow#30769]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
MacOSX 10.13.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
N/A
- TensorFlow installed from (source or binary):
binary
- TensorFlow version (use command below):
```python
tf.version.VERSION='2.0.0-dev20190716'
tf.version.GIT_VERSION='v1.12.1-6367-gb51a1b258b'
```
- Python version:
3.6.8
- Bazel version (if compiling from source):
N/A
- GCC/Compiler version (if compiling from source):
N/A
- CUDA/cuDNN version:
N/A
- GPU model and memory:
N/A

**Describe the current behavior**
`tf.__version__` raises an `AttributeError` exception.

**Describe the expected behavior**
`tf.__version__` should return the version (=`tf.version.VERSION`) to respect [PEP 396](https://www.python.org/dev/peps/pep-0396/).

**Code to reproduce the issue**
```python
import tensorflow as tf

print(tf.__version__)
```

**Other info / logs**
Here's the exception:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-d1382151fdd6> in <module>
      1 import tensorflow as tf
      2
----> 3 print(tf.__version__)

AttributeError: module 'tensorflow' has no attribute '__version__'
```

**Comment by lufol** [comment]
To me it seems `__version__` should be initialized here "/tensorflow/python/framework/versions.py" 
However if I run the code as @ageron on the tf-nightly version I get the same issue.
Here the full error log:
```---------------------------------------------------------------------------

AttributeError                            Traceback (most recent call last)

<ipython-input-1-f83c6d50081b> in <module>()
      1 import tensorflow as tf
----> 2 print(tf.__version__)

1 frames

/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/util/module_wrapper.py in __getattr__(self, name)
    169     except AttributeError as e:
    170       if not self._tfmw_public_apis:
--> 171         raise e
    172       if name not in self._tfmw_public_apis:
    173         raise e

/usr/local/lib/python3.6/dist-packages/tensorflow_core/python/util/module_wrapper.py in __getattr__(self, name)
    166   def __getattr__(self, name):
    167     try:
--> 168       attr = getattr(self._tfmw_wrapped_module, name)
    169     except AttributeError as e:
    170       if not self._tfmw_public_apis:

AttributeError: module 'tensorflow' has no attribute '__version__'
```

The process of the declaration of the version is described nicely at [stackoverflow](https://stackoverflow.com/questions/50427475/checking-tensorflow-version-in-python-tf-version-vs-tf-version?rq=1)
Maybe this helps find the source of this problem quicker.

**Comment by yongtang** [comment]
The issue is that module wrapping hides all symbols starts with `_` from `__all__` so `__version__` is hided as well. Created a PR #30778 for the fix.

**Comment by pandrey-fr** [comment]
"It's not a bug, it's a feature."

In 2.0 (at least as of now), `tf.__version__` is replaced with `tf.version.VERSION` and all other (pseudo) constants under `tf.version`, so as to make versioning clearer. This is actually indicated in the mere issue template you filled in...

**Comment by gadagashwini** [comment]
@ageron,
Issue is fixed in latest TF 2.0 nightly version. Can we close the issue. Thanks!
